sklearn/linear_model/_ridge.py

```diff
@@ -2401,10 +2397,13 @@ def fit(self, X, y, sample_weight=None, **params):
                 if sample_weight is not None:
                     routed_params.scorer.score["sample_weight"] = sample_weight
 
+            if self.scoring is None:
+                scorer = None
```

Comment:
Thanks for looking into this. I am not sure if I understand what you mean, @glemaitre.

I will try to explain what I meant:

We have to call `self._get_scorer()` in any case (even if `scorer=None`, because in this case we need to use RidgeCV's `score()` method via `Passthrougscorer`.

So the code needs to be like this:

```
scorer = self._get_scorer()  
...
if self.scoring is None:  
    scorer = None  
```

The scorer needs to be defined (`scorer = self._get_scorer()`) pretty early in `fit()`, because it is used in `process_routing()` already if metadata routing is enabled and cv is None. 

But then below we need to reset `scorer` to None if the original user-inted was to not pass a scoring: 

```
if self.scoring is None:  
    scorer = None 
```

If we would not do that, we would get an `AttributeError: '_IdentityRegressor' object has no attribute 'score'` because the scorer is already something else than None and we would pass something else than None into `_RidgeGCV()` and we go down the wrong path of the code.

Basically, we are misusing the variable `scorer` by having it have too many meanings at the same time. But maybe that's better than messing with the conventions?

I have added a comment to explain what I mean. Does it all make sense?

Reply:
Yes completely. I missed that we were going to get the combination of `_PassthroughScorer` + `_IdentityRegressor` that end-up in a non-compatible scikit-learn estimator.

I think we can merge as-is. Maybe one clean-up could be to check that we get an `_Identity...` estimator that follow the scikit-learn API but I'm not sure that we can end-up with something working.